klog_test.go

```diff
@@ -1493,6 +1597,63 @@ func (l *testLogr) WithValues(...interface{}) logr.Logger {
 	panic("not implemented")
 }
 
+type callDepthTestLogr struct {
+	testLogr
+	callDepth int
+}
+
+func (l *callDepthTestLogr) resetCallDepth() {
+	l.mutex.Lock()
+	defer l.mutex.Unlock()
+	l.callDepth = -1
+}
+
+func (l *callDepthTestLogr) WithCallDepth(depth int) logr.Logger {
+	l.mutex.Lock()
+	defer l.mutex.Unlock()
+	l.callDepth = depth
```

Comment:
Nit: this is not how `WithCallDepth` normally would be implemented (clone, then increment depth in the clone instead of setting it). Can you add a short comment explaining that this is done this way for the sake of keeping the tests simple?